src/mappings/aave.ts

```diff
@@ -1,32 +1,69 @@
-import { Address } from '@graphprotocol/graph-ts'
-import { Transfer } from '../../generated/AaveToken/AaveToken'
+import { Address, BigInt } from '@graphprotocol/graph-ts'
+import { BalanceTransfer, Burn, Mint } from '../../generated/AaveToken/AaveToken'
 import { SupplyCapChanged } from '../../generated/AavePoolConfigurator/AavePoolConfigurator'
-
 import { loadAave } from '../entities/aave'
+import { OS_TOKEN } from '../helpers/constants'
 
 export function handleSupplyCapChanged(event: SupplyCapChanged): void {
-  const aave = loadAave()!
-  aave.osTokenSupplyCap = event.params.newSupplyCap
-  aave.save()
+  if (event.params.asset.equals(OS_TOKEN)) {
+    const aave = loadAave()!
+
+    aave.osTokenSupplyCap = event.params.newSupplyCap
+    aave.save()
+  }
 }
 
-export function handleTransfer(event: Transfer): void {
-  const from = event.params.from
-  const to = event.params.to
-  const amount = event.params.value
+function mint(value: BigInt, balanceIncrease: BigInt, onBehalf: Address): void {
+  const onBehalfString = onBehalf.toHexString().toLowerCase()
+
+  const isMintingToTreasury =
+    onBehalfString != '0xb2289e329d2f85f1ed31adbb30ea345278f21bcf' &&
+    onBehalfString != '0xe8599f3cc5d38a9ad6f3684cd5cea72f10dbc383' &&
+    onBehalfString != '0xbe85413851d195fc6341619cd68bfdc26a25b928' &&
+    onBehalfString != '0x5ba7fd868c40c16f7adfae6cf87121e13fc2f7a0' &&
+    onBehalfString != '0x8a020d92d6b119978582be4d3edfdc9f7b28bf31' &&
+    onBehalfString != '0x053d55f9b5af8694c503eb288a1b7e552f590710' &&
+    onBehalfString != '0x464c71f6c2f760dda6093dcb91c24c39e5d6e18c'
 
-  if (amount.isZero()) {
-    return
+  if (isMintingToTreasury) {
+    const aave = loadAave()!
+    const userBalanceChange = value.minus(balanceIncrease)
+
+    aave.osTokenTotalSupplied = aave.osTokenTotalSupplied.plus(userBalanceChange)
+    aave.save()
   }
+}
 
+function burn(value: BigInt, balanceIncrease: BigInt): void {
   const aave = loadAave()!
+  const userBalanceChange = value.plus(balanceIncrease)
 
-  if (from.equals(Address.zero())) {
-    aave.osTokenTotalSupplied = aave.osTokenTotalSupplied.plus(amount)
-  }
-  if (to.equals(Address.zero())) {
-    aave.osTokenTotalSupplied = aave.osTokenTotalSupplied.minus(amount)
+  aave.osTokenTotalSupplied = aave.osTokenTotalSupplied.minus(userBalanceChange)
+  aave.save()
+}
+
+export function handleMint(event: Mint): void {
+  mint(event.params.value, event.params.balanceIncrease, event.params.onBehalfOf)
+}
+
+export function handleBurn(event: Burn): void {
+  burn(event.params.value, event.params.balanceIncrease)
+}
+
+let RAY = BigInt.fromI32(10).pow(27)
+let halfRAY = RAY.div(BigInt.fromI32(2))
```

Comment:
the constants should be defined at the top

Reply:
Fixed https://github.com/stakewise/v3-subgraph/commit/d1b0f7dc3e3bc1a9ffd34544c24af6e14b8cee4a